process/hyper_block_processor.go

```diff
@@ -51,8 +90,60 @@ func (hbp *hyperBlockProcessor) Process(hyperBlock *hyperBlock.HyperBlock) (*sch
 		return nil, err
 	}
 
-	avroHyperBlock.ShardBlocks = shardBlocks
-	avroHyperBlock.Transactions = txs
-	avroHyperBlock.EpochStartInfo = epochStartInfo
-	return avroHyperBlock, nil
+	return &schemaV2.HyperBlock{
+		Hash:                   hash,
+		PrevBlockHash:          prevBlockHash,
+		StateRootHash:          stateRootHash,
+		Nonce:                  int64(hyperBlock.Nonce),
+		Round:                  int64(hyperBlock.Round),
+		Epoch:                  int32(hyperBlock.Epoch),
+		NumTxs:                 int32(hyperBlock.NumTxs),
+		AccumulatedFees:        accumulatedFees,
+		DeveloperFees:          developerFees,
+		AccumulatedFeesInEpoch: accumulatedFeesInEpoch,
+		DeveloperFeesInEpoch:   developerFeesInEpoch,
+		Timestamp:              int64(hyperBlock.Timestamp),
+		EpochStartInfo:         epochStartInfoOrNil(epochStartInfo),
+		ShardBlocks:            shardBlocksOrNil(shardBlocks),
+		Transactions:           txsOrNil(txs),
+		StateChanges:           nil,
+		Status:                 hyperBlock.Status,
+	}, nil
+}
+
+func txsOrNil(txs []*schemaV2.Transaction) []*schemaV2.Transaction {
+	if len(txs) == 0 {
+		return nil
+	}
+	return txs
+}
+
+func shardBlocksOrNil(shardBlocks []*schemaV2.ShardBlocks) []*schemaV2.ShardBlocks {
+	if len(shardBlocks) == 0 {
+		return nil
+	}
+	return shardBlocks
+}
+
+func epochStartInfoOrNil(epochStartInfo *schemaV2.EpochStartInfo) *schemaV2.EpochStartInfo {
+	if emptyEpochStartInfo(epochStartInfo) {
+		return nil
+	}
+
+	return epochStartInfo
+}
+
+func emptyEpochStartInfo(epochStartInfo *schemaV2.EpochStartInfo) bool {
```

Comment:
`isEmpty...` (renaming, optional)

Reply:
Right, renamed it to `isEpochStartInfoEmpty`